src/encoded/static/components/dataset.js

```diff
@@ -1903,9 +1903,15 @@ function sortStage(a, b) {
         return sortMouseArray(a.life_stage_age, b.life_stage_age);
     }
     // special case with multiple replicates (old data)
-    if (a.replicates[0].library.biosample.age_display && b.replicates[0].library.biosample.age_display) {
+    if (a.replicates && b.replicates && a.replicates[0].library.biosample.age_display && b.replicates[0].library.biosample.age_display) {
         return (a.replicates[0].library.biosample.age_display.split(' ')[0] - b.replicates[0].library.biosample.age_display.split(' ')[0]);
     }
+    if (a.replicates && a.replicates[0].library.biosample.age_display) {
+        return -a.replicates[0].library.biosample.age_display.split(' ')[0];
+    }
+    if (b.replicates && b.replicates[0].library.biosample.age_display) {
+        return b.replicates[0].library.biosample.age_display.split(' ')[0];
+    }
```

Comment:
For these you can also do the very weird-looking `a.replicates?.[0].library.biosample.age_display` to reduce the &&.

Not all replicates have libraries, and not all libraries have biosamples. Do you also need to check for their existence before dereferencing them, or are those already guaranteed by the time you get to this code?

Reply:
I'm not sure about this, maybe @emro would have a better idea. However, I did try checking for library and biosample on my branch and it somehow ends up breaking the UI again. e.g. `b.replicates && b.replicates[0].library?.biosample?.age_display`. Perhaps I did the check incorrectly though.